src/gpio_monitor.cpp

```diff
@@ -1,31 +1,132 @@
 #include "gpio_monitor.hpp"
 
+#include "constants.hpp"
 #include "logger.hpp"
 #include "types.hpp"
+#include "utility/dbus_utility.hpp"
 #include "utility/json_utility.hpp"
 
+#include <boost/asio.hpp>
+#include <boost/bind/bind.hpp>
+#include <gpiod.hpp>
+
 namespace vpd
 {
 bool GpioEventHandler::getPresencePinValue()
 {
-    // ToDo Add implementation.
-    return true;
+    gpiod::line l_presenceLine = gpiod::find_line(m_presencePin);
```

Comment:
Instead of this API can we make use of json utility function to process gpio?
That API will take care of all the error handling for you.

Reply:
Cannot use the json utility function processGpioPresenceTag(), which expects baseAction (i.e preAction) and flagToProcess (i.e collection) tags but in the JSON for polling we have only one tag "pollingRequired".

Reply:
Oh okay, that is why we should have kept the same level of tags in JSON so that we could have made use of existing functions. 
Else we have to duplicate code unnecessarily.
Anyways lets se what others have to say, lets take call,
-> either make the level same in JSON or,
-> duplicate the code.

Anyways in the later case, you will have to pull all the error handling again here.